From 333cf12e509771c57d12018645becf2d4386023c Mon Sep 17 00:00:00 2001
From: Meng Li <Meng.Li@windriver.com>
Date: Thu, 29 Feb 2024 14:06:20 +0800
Subject: [PATCH] arm: dts: k3: binman: create a symlink file pointing to R5
 secure bootloader

When secure boot feature is enabled, this patch should be used to create
a tiboot3.bin symlink file that can be recognized by bootrom. In this way,
the secure bootloader will be deployed in boot device and run on TI-AM62Ax
HS-SE(High Security-Security Enforced) device. In addition, remove the symlink
file for TI-AM62Ax HS-FS(High Security-Field Securable) device.

Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/dts/k3-am62a-sk-binman.dtsi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/dts/k3-am62a-sk-binman.dtsi b/arch/arm/dts/k3-am62a-sk-binman.dtsi
index eb1f7a929f..9819472b27 100644
--- a/arch/arm/dts/k3-am62a-sk-binman.dtsi
+++ b/arch/arm/dts/k3-am62a-sk-binman.dtsi
@@ -14,6 +14,7 @@
 &binman {
 	tiboot3-am62ax-hs-evm.bin {
 		filename = "tiboot3-am62ax-hs-evm.bin";
+		symlink = "tiboot3.bin";
 		ti-secure-rom {
 			content = <&u_boot_spl>, <&ti_fs_enc>, <&combined_tifs_cfg>,
 				<&combined_dm_cfg>, <&sysfw_inner_cert>;
@@ -59,7 +60,6 @@
 &binman {
 	tiboot3-am62ax-hs-fs-evm.bin {
 		filename = "tiboot3-am62ax-hs-fs-evm.bin";
-		symlink = "tiboot3.bin";
 		ti-secure-rom {
 			content = <&u_boot_spl_fs>, <&ti_fs_enc_fs>, <&combined_tifs_cfg_fs>,
 				<&combined_dm_cfg_fs>, <&sysfw_inner_cert_fs>;
-- 
2.34.1

